django/db/models/sql/query.py

```diff
@@ -433,6 +450,7 @@ def get_aggregation(self, using, aggregate_exprs):
             isinstance(self.group_by, tuple)
             or self.is_sliced
             or has_existing_aggregation
+            or refs_subquery
```

Comment:
Missed optimization barrier that forces the usage of a subquery pushdown (doing `SELECT agg0 FROM (SELECT ...)`) if any of the aggregates refer to a subquery.